src/main/java/uk/gov/justice/hmpps/offenderevents/services/EventRetrievalService.java

```diff
@@ -79,4 +83,56 @@ public void pollEvents(final LocalDateTime currentTime) {
             log.warn("Skipping Event Retrieval as start after end, start = {}, end = {}", startTime, endTime);
         }
     }
+
+    /**
+     * For testing run a poll for the immediately ended minute, repeat 1 min later and compare:
+     * <ol>
+     *  <li>run for the latest minute (POLL_NAME_TEST.time -> now - 1 second) remembering time range and count, A
+     *  <li>run for [PREVIOUS_POLL_NAME_TEST -> POLL_NAME_TEST] time range remembering time range and count, B
+     *  <li>compare count B with POLL_NAME_TEST.count - should be the same
+     *  <li>store count A in POLL_NAME_TEST.count
+     *  <li>move POLL_NAME_TEST.time to PREVIOUS_POLL_NAME_TEST.time
+     *  <li>store A end time in POLL_NAME_TEST.time
+     * </ol>
+     */
+    public void runTestPolls(final LocalDateTime now) {
+        final LocalDateTime endTimeA = now.minus(1, SECONDS);
+
+        repository.findById(POLL_NAME_TEST).ifPresentOrElse(
+                test -> {
+                    final var previousTest = repository.findById(PREVIOUS_POLL_NAME_TEST).orElseThrow();
+                    final var eventsA = externalApiService.getEvents(test.getNextStartTime(), endTimeA);
+                    final var countA = eventsA.size();
+                    log.debug("runTestPolls(): A interval {} to {}, count {}", test.getNextStartTime(), endTimeA, countA);
```

Comment:
wondering if you need to log `A` here, won't it be the same as the next `B` record?  If it isn't then the `warn` block will execute so not sure it adds anything (and doubles the number of log entries every minute)